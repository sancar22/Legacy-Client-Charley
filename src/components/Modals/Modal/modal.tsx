import React, { ChangeEvent, FormEvent, useState } from 'react';
import { Link } from 'gatsby';
import { trackPromise } from 'react-promise-tracker';

import LoadingInd from '../../LoadingInd/loadingInd';
import apiService from '../../../services/apiService';

import * as styles from './modal.module.css';

const Modal = ({
  show,
  handleClose,
}: {
  show: boolean;
  handleClose: () => void;
}): JSX.Element => {
  const [url, setUrl] = useState('');
  const [error, setError] = useState(false);
  const [success, setSuccess] = useState(false);

  const handleChange = (event: ChangeEvent<HTMLInputElement>): void => {
    setUrl(event.target.value);
    setError(false);
    setSuccess(false);
  };
  const handleSubmit = (event: FormEvent<HTMLFormElement>): void => {
    event.preventDefault();

    trackPromise(
      apiService.scrapeRecipe(url).then((res) => {
        if (res.ok) {
          setUrl('');
          setSuccess(true);
          setError(false);
        } else {
          setError(true);
          setSuccess(false);
        }
      }),
    );
  };

  return (
    <div className={show ? styles.modalShow : styles.modalHide}>
      <div className={styles.modalMain}>
        <p className={styles.text__header}>Add your url</p>

        <form className={styles.form} onSubmit={handleSubmit}>
          <input
            name='url'
            value={url}
            onChange={handleChange}
            className={styles.input}
          />
          <LoadingInd color='lightgrey' />
          {success && (
            <div className={styles.text__success}>
              succcess! take me to <Link to='/recipes'>my recipes</Link>
            </div>
          )}
          {error && (
            <p className={styles.text__error}>
              unable to import, please try another website
            </p>
          )}

          <button type='submit' disabled={!url}>
            submit
          </button>
        </form>

        <button onClick={handleClose} className={styles.close}>
          x
        </button>
      </div>
    </div>
  );
};

export default Modal;
