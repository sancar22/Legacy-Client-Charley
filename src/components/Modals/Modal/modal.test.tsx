import React from 'react';
import { create } from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import reduxStore from '../../../state/store';
import Modal from './modal';

const { store } = reduxStore();

describe('Modal component', () => {
  const modal = create(
    <Provider store={store}>
      <Modal show={true} handleClose={jest.fn()} />
    </Provider>,
  );
  test('match previous Modal snapshot', () => {
    expect(modal).toMatchSnapshot();
  });
});
