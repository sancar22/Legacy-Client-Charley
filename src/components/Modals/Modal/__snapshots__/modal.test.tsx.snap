// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component match previous Modal snapshot 1`] = `
<div>
  <div>
    <p>
      Add your url
    </p>
    <form
      onSubmit={[Function]}
    >
      <input
        name="url"
        onChange={[Function]}
        value=""
      />
      <button
        disabled={true}
        type="submit"
      >
        submit
      </button>
    </form>
    <button
      onClick={[MockFunction]}
    >
      x
    </button>
  </div>
</div>
`;
